### PR TITLE
Modification to remove some warnings

### PR DESCRIPTION
**Pull request by Diogo-Rossi**
Modification to remove warning when `Slot` is used as decorator of a method inside a class.

The warning is issued by `pylance` language server (that used Pyright in VS Code)

image

Message of the issue:
```
Argument of type "(self: Self@MainWindowVM, somestring: Unknown) -> None" cannot be assigned to parameter of type "(str) -> RetT@Slot"
  Type "(self: Self@MainWindowVM, somestring: Unknown) -> None" cannot be assigned to type "(str) -> RetT@Slot"
    Parameter 1: type "str" cannot be assigned to type "Self@MainWindowVM"
      "str" is incompatible with "MainWindowVM"
    Function accepts too few positional parameters; expected 2 but received 1
```
___

Modification to remove 2 warnings in `SIGNAL` and `connect`

image

Message:

```
Argument of type "str" cannot be assigned to parameter "arg__1" of type "bytes" in function "connect"
  "str" is incompatible with "bytes"Pylance[reportGeneralTypeIssues]
Argument of type "Literal['clicked()']" cannot be assigned to parameter "arg__1" of type "bytes" in function "SIGNAL"
  "Literal['clicked()']" is incompatible with "bytes"Pylance[reportGeneralTypeIssues]
```

### CODE REVIEW

**Comment by bluebird75** [comment]
Hi,

Thanks for the contribution.

The `str` argument  makes sense.

I am questionning the `bytes` argument part, maybe it should just go. I have not read that documentation recently but I have my doubts just be looking at the stubs.

Do you have any idea when the `bytes` argument is correct ? Because PySide2 generated stubs have many inconsistencies like this one.

**Comment by bluebird75** [comment]
Ok, I did some checks. The `bytes` argument is clearly a mistake.

```
>>> from PySide2.QtCore import SIGNAL
>>> SIGNAL('abc')
'2abc'
>>> SIGNAL(b'abc')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: 'PySide2.QtCore.SIGNAL' called with wrong argument values:
  PySide2.QtCore.SIGNAL(b'abc',)
Found signature:
  PySide2.QtCore.SIGNAL(bytes)
>>> from PySide2.QtCore import SLOT
>>> SLOT('abc')
'1abc'
>>> SLOT(b'abc')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: 'PySide2.QtCore.SLOT' called with wrong argument values:
  PySide2.QtCore.SLOT(b'abc',)
Found signature:
  PySide2.QtCore.SLOT(bytes)
>>>
```

Can you update you update your patch to change the bytes arguments to str ?

If you feel like writing a test, it's also welcome. See the README for some indications.

**Comment by bluebird75** [comment]
Thanks for using PySide2 stubs by the way. Do not hesitate to report any further problem, even minors.

**Comment by Diogo-Rossi** [comment]
> Can you update you update your patch to change the bytes arguments to str ?
Ok!

File: PySide2-stubs/QtCore.pyi [hunk 1]
```diff
@@ -13779,6 +13779,10 @@ Func_args_T1_T2_T3_T4_T5_T6_returns_RetT = typing.Callable[[T1, T2, T3, T4, T5,
 ###        If result is provided, it falls back to f(*Any) -> result
 ###        For 7 arguments, only the return value is typechecked if the result is provided.
 
+@typing.overload
+def Slot(*some_types: typing.Type, result: typing.Type = ...) \
+        -> typing.Callable[[typing.Callable[..., RetT]],
+                           typing.Callable[..., RetT]]: ...
 @typing.overload
```
[the remark this answers is not in the document]
Comment (Diogo-Rossi) [reply PySide2-stubs/QtCore.pyi:13786]: This "*change in order of overload*" is to correct the first issue mentioned in the PR:

> Modification to remove warning when `Slot` is used as decorator of a method inside a class.
> 
> The warning is issued by `pylance` language server (that used Pyright in VS Code)
> 
> image
> 
> Message of the issue:
> 
> ```
> Argument of type "(self: Self@MainWindowVM, somestring: Unknown) -> None" cannot be assigned to parameter of type "(str) -> RetT@Slot"
>   Type "(self: Self@MainWindowVM, somestring: Unknown) -> None" cannot be assigned to type "(str) -> RetT@Slot"
>     Parameter 1: type "str" cannot be assigned to type "Self@MainWindowVM"
>       "str" is incompatible with "MainWindowVM"
>     Function accepts too few positional parameters; expected 2 but received 1
> ```

I added this [test](https://github.com/Diogo-Rossi/PySide2-stubs/blob/test-slot-inside-class/tests/slotInsideClass.py) on [this branch](https://github.com/Diogo-Rossi/PySide2-stubs/tree/test-slot-inside-class) that shows this warning when use `Slot` inside a class (But, maybe it is a Pylance Issue).

**Comment by bluebird75** [comment]
Thanks

**Comment by bluebird75** [comment]
By the way, do you have an example code of using SIGNAL, Slot and connect ? I want to add this to the test suite

**Pull request merged**